modules/exploits/linux/http/multi_ncc_ping_exec.rb

```diff
@@ -0,0 +1,204 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Exploit::Remote
+  Rank = ManualRanking  #only tested in emulated environment
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer::HTML
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'D-Link/TRENDnet NCC Command Injection (ping.ccp)',
+      'Description'    => %q{
+        This module exploits a remote command injection vulnerability on several routers.
+        This module was tested in an emulated environment of a DIR-626L only. Several
+        D-Link and TRENDnet devices are reported as affected:
+        D-Link DIR-626L (Rev A) - v1.04b04,
+        D-Link DIR-636L (Rev A) - v1.04,
+        D-Link DIR-808L (Rev A) - v1.03b05,
+        D-Link DIR-810L (Rev A) - v1.01b04,
+        D-Link DIR-810L (Rev B) - v2.02b01,
+        D-Link DIR-820L (Rev A) - v1.02B10,
+        D-Link DIR-820L (Rev A) - v1.05B03,
+        D-Link DIR-820L (Rev B) - v2.01b02,
+        D-Link DIR-826L (Rev A) - v1.00b23,
+        D-Link DIR-830L (Rev A) - v1.00b07,
+        D-Link DIR-836L (Rev A) - v1.01b03,
+        TRENDnet TEW-731BR (Rev 2) - v2.01b01
+      },
+      'Author'         =>
+        [
+          'Peter Adkins <peter.adkins[at]kernelpicnic.net>', # Vulnerability discovery and initial PoC
+          'Tiago Caetano Henriques', # Vulnerability discovery and initial PoC
+          'Michael Messner <devnull[at]s3cur1ty.de>', # Metasploit module
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['CVE', '2015-1187'],
+          ['BID', '72816'],
+          ['URL', 'https://github.com/darkarnium/secpub/tree/master/Multivendor/ncc2'], #advisory with PoC
+          ['URL', 'http://seclists.org/fulldisclosure/2015/Mar/15'], #advisory with PoC
+          ['URL', 'http://securityadvisories.dlink.com/security/publication.aspx?name=SAP10052'] #vendor site with update
+        ],
+      'Targets'        =>
+        [
+          [ 'Linux mipsel Payload',
```

Comment:
mipsel sounds good to me:

```MIPSel refers to a MIPS architecture using a little endian byte order``` from http://en.wikipedia.org/wiki/MIPS_instruction_set

Also, called mipsel by debian (prolly others): https://www.debian.org/ports/mips/